src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -80,10 +89,43 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if ($limit = $input->getOption('limit')) {
-            $receiver = new MaximumCountReceiver($receiver, $limit);
+            $receiver = new StopWhenMessageCountIsExceededReceiver($receiver, $limit, $this->logger);
+        }
+
+        if ($memoryLimit = $input->getOption('memory-limit')) {
+            $receiver = new StopWhenMemoryUsageIsExceededReceiver($receiver, $this->convertToBytes($memoryLimit), $this->logger);
         }
 
         $worker = new Worker($receiver, $this->bus);
         $worker->run();
     }
+
+    private function convertToBytes($memoryLimit)
+    {
+        if ('-1' === $memoryLimit) {
```

Comment:
I really don't get the point of this `-1` here. I appreciate it's coming from a copy/pasted code somewhere else but I'd remove it.

Reply:
I agree